src/main/core/manager.c

```diff
@@ -646,6 +653,55 @@ static void _manager_heartbeat(Manager* manager, SimulationTime simClockNow) {
     }
 }
 
+static void _manager_checkResourceUsage(Manager* manager) {
+    if (manager->checkFdUsage) {
+        unsigned long fd_count = 0;
+        DIR* dirp = opendir("/proc/self/fd");
+        if (dirp != NULL) {
+            struct dirent* entry = NULL;
+            while ((entry = readdir(dirp)) != NULL) {
+                fd_count++;
+            }
+        } else {
+            warning("Unable to open '/proc/self/fd': %s", g_strerror(errno));
+            manager->checkFdUsage = false;
+            return;
+        }
+
+        struct rlimit fd_lim;
+        if (getrlimit(RLIMIT_NOFILE, &fd_lim) != 0) {
+            warning("Unable to get fd limit: %s", g_strerror(errno));
+            manager->checkFdUsage = false;
+            return;
+        }
+
+        if (fd_count > fd_lim.rlim_cur * 90 / 100) {
```

Comment:
I don't remember whether it's actually needed here, but parens would be helpful to the reader either way :). `fd_count > (fd_lim.rlim_cur * 90 / 100)`

Reply:
Added.